crates/core/src/repository/backend/git.rs

```diff
@@ -193,70 +226,78 @@ mod github {
         remote_packages: &mut HashMap<ReleaseChannel, RemotePackage>,
         release_channel: ReleaseChannel,
         release: &Release,
     ) -> Result<(), RepositoryError> {
-        let num_non_classic = release
-            .assets
-            .iter()
-            .filter(|a| a.name.ends_with("zip"))
-            .filter(|a| !a.name.to_lowercase().contains("classic"))
-            .count();
-
-        let num_classic = release
-            .assets
-            .iter()
-            .filter(|a| a.name.ends_with("zip"))
-            .filter(|a| a.name.to_lowercase().contains("classic"))
-            .count();
-
-        if flavor.base_flavor() == Flavor::Retail && num_non_classic > 1
-            || flavor.base_flavor() == Flavor::ClassicEra && num_classic == 0 && num_non_classic > 1
+        // Check if release has a `release.json` file, else use fallback solution..
+        let asset: Result<Option<&ReleaseAsset>, serde_json::Error> = if let Some(release_file) =
+            release.assets.iter().find(|a| a.name == "release.json")
         {
-            return Err(RepositoryError::GitIndeterminableZip {
-                count: num_non_classic,
-                url: url.to_string(),
-            });
-        } else if flavor.base_flavor() == Flavor::ClassicEra && num_classic > 1 {
-            return Err(RepositoryError::GitIndeterminableZipClassic {
-                count: num_classic,
-                url: url.to_string(),
-            });
-        }
+            // If we find `release.json`, we download content, and parse it.
+            let mut resp = request_async(&release_file.browser_download_url, vec![], None).await?;
+            let release_file: ReleaseFile = resp.json().await?;
 
-        let asset = release
-            .assets
-            .iter()
-            .find(|a| {
-                if flavor.base_flavor() == Flavor::Retail {
-                    a.name.ends_with("zip") && !a.name.to_lowercase().contains("classic")
-                } else if num_classic > 0 {
-                    a.name.ends_with("zip") && a.name.to_lowercase().contains("classic")
-                } else {
+            // Try to find the package, which contains the flavor we are looking for.
+            if let Some(release_file) = release_file
+                .releases
+                .iter()
+                .find(|r| r.metadata.iter().any(|m| m.flavor == flavor))
+            {
+                // Find the asset, based on what we know from the `release.json`.
+                let asset = release
+                    .assets
+                    .iter()
+                    .find(|a| a.name == release_file.filename);
+                Ok(asset)
+            } else {
+                Ok(None)
+            }
+        } else {
+            // Fallback solution where we try to look at the asset name to determine the correct file.
+            // Eg:
+            // `foobar-classic` => ClassicEra file.
+            // `foobar-bcc` => ClassicTbc file.
```

Comment:
This fallback solution has one downside compared to the old method... For classic / bcc, if no `classic` or `bcc` named file exists, respectively, but a single `.zip` asset exists, it won't use that.

Reply:
Most authors should conform to the new `release.json` or multi-asset naming convention, so I'm fine taking a stance against this and just informing authors of how to fix their releases.

Reply:
Agree, all repos which is using BigWigs packager will get this file, so I would assume it will become defacto standard.